### DISCUSSION TITLE
Integrate themelint into helix so that you don't need to clone the repository

### DISCUSSION THREAD

**Discussion by ProfDoof** [helix-editor/helix#8379] [Ideas]
Hi! So, I'm starting to use helix on one of my machines as a way to force myself to focus but I didn't really find any themes that fit my desires exactly. So I'm trying to write my own. However, currently when you want to write a theme and lint it you either need to repeatedly load it in through the theme command or you need to clone the entire repository and build the entire tool to use the `xtask themelint` mentioned in the documentation. This isn't exactly ideal. I have a couple of suggestions surrounding this but my baseline would be integrate the tool into hx directly somehow (maybe this is being considered when the plugins system is finished?) so that I can lint my theme as I develop it. 

As an aside, it would be ideal if there was a reference somewhere of what each theme key impacts visually. I'm finding it difficult to predict what the changes I'm making will do since I can't always see the changes as I don't have the relevant thing open to see the change and I don't know what to open to see that. It also could just be good to have a single theme impact reference file that you can just load in and see all the impacts of the changes you are making. Once I figure out how to do multiple windows I may even try to write something like that unless someone already has an example of what that looks like. Because the examples I saw in the wiki are incomplete (for example, many of them don't show what a breakpoint looks like with that theme).

**Comment by kirawi** [comment]
The reference is at https://docs.helix-editor.com/themes.html#scopes

**Reply by ProfDoof** [reply]
So, I have been referencing that but I more meant a visual representation of the scopes that allows you to directly see what each of the keys will impact. Something like a screenshot of the editor with each thing labeled or a file that demos all of the various keys that are impacted by a theme allowing you to easily visualize the impact of every change. This also allows you to better identify how well the colors contrast in a simple manner. The only other thing I would improve there would be some automated suggestions around the contrast levels and accessibility score of a theme.

**Comment by pascalkuthe** [comment]
I don't think we will integrate thenelint into helix itself. @archseer didn't even want to merge themelint at all and it was only merged due to a miscommunication.

**Reply by ProfDoof** [reply]
I'll be honest, I don't really follow. Was it just not wanting to incur the maintenance cost?
